client.go

```diff
@@ -192,6 +199,10 @@ func (c *Client) exchange(m *Msg, a string) (r *Msg, rtt time.Duration, err erro
 	// Otherwise use the client's configured UDP size.
 	if opt == nil && c.UDPSize >= MinMsgSize {
 		co.UDPSize = c.UDPSize
+		// In this case also set the the EDNS0 bufsize to allow for larger responses.
+		if c.SetUDPSize {
+			m.SetEdns0(c.UDPSize, true /* dnssec */)
```

Comment:
Setting dnssec to true seems like an unexpected side effect of enabling larger payloads.  It might result in even larger responses that don't fit into a UDP response.  If nothing else I think this behavior should be noted in the doc string for SetUDPSIze.